File: 424.LongestRepeatingCharacterReplacement.java
class Solution {
    public int characterReplacement(String s, int k) {
        int[] arr = new int[26];
        int ans = 0;
        //int max = 0;
        int l = 0;

        for (int r = 0; r < s.length(); r++){
            arr[s.charAt(r) - 'A']++;

            //max = Math.max(max, arr[s.charAt(r) - 'A']);
            //while (r - l + 1 - max > k)(
            while (r - l + 1 - getMaxOfArray(arr) > k){
                arr[s.charAt(l) - 'A']--;
                l++;
            }

            ans = Math.max(ans, r - l + 1);
        }
        return ans;
    }

    private int getMaxOfArray(int[] arr){
        int max = arr[0];
        for (int i = 1; i < arr.length; i++){
            max = Math.max(max, arr[i]);
        }

        return max;
    }
}

File: 424.LongestRepeatingCharacterReplacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        d = {}
        l = 0
        res = 0
        r = 0
        while r < len(s):
            d[s[r]] = 1 + d.get(s[r], 0)

            while (r - l + 1) - max(d.values()) > k:
                d[s[l]] -= 1
                l += 1
            res = max(res, r - l + 1) 
            r += 1
            
        return res


        
